benchmark/fs/bench-opendir.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const common = require('../common');
+const fs = require('fs');
+const path = require('path');
+
+const bench = common.createBenchmark(main, {
+  n: [100],
+  dir: [ 'lib', 'test/parallel'],
+  mode: [ 'async', 'sync' ]
+});
+
+function main({ n, dir, mode }) {
+  const fullPath = path.resolve(__dirname, '../../', dir);
+
+  (async () => {
```

Comment:
Nit: what about using an `async` main function? I am not sure but I think we have no contract that prohibits the main function to return a promise?

Reply:
Yeah, that makes sense – I’ll try it once I’m done with other things I’m working on :)